eo-runtime/src/main/java/EOorg/EOeolang/EOerror.java

```diff
@@ -126,5 +127,29 @@ public ExError(final Phi enclosure) {
         public Phi enclosure() {
             return this.enc;
         }
+
+        /**
+         * Retrieve message from enclosure safely.
+         * @param enclosure Enclosure.
+         * @return String message.
+         * @checkstyle IllegalCatchCheck (20 lines)
+         */
+        private static String safeMessage(final Phi enclosure) {
+            String result;
+            if (enclosure == null) {
+                result = "null Phi";
+            } else {
+                try {
+                    result = new PhContainingUTF8(enclosure).toString();
```

Comment:
@levBagryansky if this class is an instance of `Phi` _and_ it makes phi objects safer by properly printing their `toString()`, why do we use only here, in this private method? Why don't we decorate _all_ our phi objects with this decorator?

Reply:
@yegor256 This class is correct to use in order to print bytes of `data` as UTF8 String. So `PhContainingUTF8(EOint)` produces something strange. We can also use this class in `Data.ToPhi(java.util.String)`

Reply:
@levBagryansky maybe we shouldn't call this class `Phi...` and should not make it a decorator of `Phi`? It doesn't look like a decorator to me, but a "smart" class: https://www.yegor256.com/2016/04/26/why-inputstream-design-is-wrong.html It just takes an existing instance of `Phi`, calls its method and then post-process the returned data. It doesn't _become_ phi, it only is a client of Phi.